src/Illuminate/Mail/Transport/PostmarkTransport.php

```diff
@@ -0,0 +1,296 @@
+		}
+		if ($reply_to = $message->getReplyTo())
+		{
+			$payload['ReplyTo'] = join(',', $this->convertEmailsArray($reply_to));
+		}
+		if ($bcc = $message->getBcc())
+		{
+			$payload['Bcc'] = join(',', $this->convertEmailsArray($bcc));
+		}
+	}
+
+	/**
+	 * Applies the message parts and attachments
+	 * into the API Payload.
+	 *
+	 * @param  array              $payload
+	 * @param  Swift_Mime_Message $message
+	 * @return object
+	 */
+	private function processMessageParts(&$payload, $message)
+	{
+		//Get the primary message.
+		switch ($message->getContentType())
+		{
+			case 'text/html':
+			case 'multipart/alternative':
+				$payload['HtmlBody'] = $message->getBody();
+				break;
+			default:
+				$payload['TextBody'] = $message->getBody();
+				break;
+		}
+
+		// Provide an alternate view from the secondary parts.
+		if ($plain = $this->getMIMEPart($message, 'text/plain'))
+		{
+			$payload['TextBody'] = $plain->getBody();
+		}
+		if ($html = $this->getMIMEPart($message, 'text/html'))
+		{
+			$payload['HtmlBody'] = $html->getBody();
+		}
+		if ($message->getChildren())
+		{
+			$payload['Attachments'] = array();
+			foreach ($message->getChildren() as $attachment)
+			{
+				if (is_object($attachment) and $attachment instanceof \Swift_Mime_Attachment)
+				{
+					$payload['Attachments'][] = array(
+						'Name'        => $attachment->getFilename(),
+						'Content'     => base64_encode($attachment->getBody()),
+						'ContentType' => $attachment->getContentType(),
+					);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Applies the headers into the API Payload.
+	 *
+	 * @param  array              $payload
+	 * @param  Swift_Mime_Message $message
+	 * @return object
+	 */
+	private function processHeaders(&$payload, $message)
+	{
+		if ($message->getHeaders())
+		{
+			$headers = [];
+			foreach ($message->getHeaders()->getAll() as $key => $value)
+			{
+				$fieldName = $value->getFieldName();
+
+				if ($fieldName != 'Subject' &&
+					$fieldName != 'Content-Type' &&
+					$fieldName != 'MIME-Version' &&
+					$fieldName != 'Date'
+				)
+				{
+
+					if ($value instanceof \Swift_Mime_Headers_UnstructuredHeader ||
+						$value instanceof \Swift_Mime_Headers_OpenDKIMHeader)
+					{
+						array_push($headers, [
+							"Name"  => $fieldName,
+							"Value" => $value->getValue(),
+						]);
+					}
+					else if ($value instanceof \Swift_Mime_Headers_DateHeader ||
+						$value instanceof \Swift_Mime_Headers_IdentificationHeader ||
+						$value instanceof \Swift_Mime_Headers_ParameterizedHeader ||
+						$value instanceof \Swift_Mime_Headers_PathHeader)
+					{
+						if ($value->getFieldName() != 'Subject')
+						{
+							array_push($headers, [
+								"Name"  => $fieldName,
+								"Value" => $value->getFieldBody(),
+							]);
+						}
+						if ($value->getFieldName() == 'Message-ID')
+						{
+							array_push($headers, [
+								"Name"  => 'X-PM-KeepID',
+								"Value" => 'true',
```

Comment:
Have you see *any* code in Laravel indented so far?

This whole thing needs a big refactor.